subprojects/gradle/namespaced-resources-fixer/src/test/kotlin/com/avito/android/plugin/FileResourcesFixerTest.kt

```diff
@@ -0,0 +1,147 @@
+        assertThat(result).isEqualTo(
+            """
+            class Screen {}
+        """
+        )
+    }
+
+    @Test
+    fun `change nothing - not used import`() {
+        val app = AndroidModule("com.avito.android.R", emptyList())
+        val libraries = emptyList<AndroidModule>()
+        val result = fix(
+            """
+            import com.avito.android.R
+            class Screen {}
+        """,
+            app, libraries
+        )
+        assertThat(result).isEqualTo(
+            """
+            import com.avito.android.R
+            class Screen {}
+        """
+        )
+    }
+
+    @Test
+    fun `change nothing - ambiguous imports`() {
+        val app = AndroidModule("com.avito.android.R", emptyList())
+        val libraries = listOf(
+            AndroidModule("com.avito.android.a", listOf("container")),
+            AndroidModule("com.avito.android.b", listOf("container"))
+        )
+        val result = fix(
+            """
+            import com.avito.android.R
+
+            class Screen {
+                val id = R.id.container
+            }
+        """,
+            app, libraries
+        )
+        assertThat(result).isEqualTo(
+            """
+            import com.avito.android.R
+
+            class Screen {
+                val id = R.id.container
+            }
+        """
+        )
+    }
+
+    @Test
+    fun `replace library id - simple property`() {
+        val app = AndroidModule("com.avito.android.R", emptyList())
+        val libraries = listOf(
+            AndroidModule("com.avito.android.library", listOf("container"))
+        )
+        val result = fix(
+            """
+            import com.avito.android.R
+
+            class Screen {
+                val id = R.id.container
+            }
+        """.trimIndent(),
+            app, libraries
+        )
+        assertThat(result).isEqualTo(
+            """
+            import com.avito.android.R
+            import com.avito.android.library.R as library_R
+
+            class Screen {
+                val id = library_R.id.container
+            }
+        """.trimIndent()
+        )
+    }
+
+    @Test
+    fun `replace library id - function argument`() {
+        val app = AndroidModule("com.avito.android.R", emptyList())
+        val libraries = listOf(
+            AndroidModule("com.avito.android.library", listOf("icon"))
+        )
+        val result = fix(
+            """
+            import androidx.test.espresso.matcher.ViewMatchers
+            import com.avito.android.R
+            import com.avito.android.test.page_object.ViewElement
+
+            class Element() : ViewElement {
+                val icon = element<ViewElement>(ViewMatchers.withId(R.id.icon))
+            }
+        """.trimIndent(),
+            app, libraries
+        )
+        assertThat(result).isEqualTo(
+            """
+            import androidx.test.espresso.matcher.ViewMatchers
+            import com.avito.android.R
```

Comment:
The same ^

Reply:
https://github.com/avito-tech/avito-android/pull/510#discussion_r459052243
Answered there